department-of-veterans-affairs/va.gov-cms
# VACMS-12568: Smart Date upgrade for drupal 10

## Description
Rolling in smart_date upgrade work from #12568 into D10 work.

Relates to #12568
Relates to #12252 

## QA steps
![222512801-20188eb2-84ba-4462-9293-814ced5349e2](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/85581471/44d7c0c4-ba0f-4557-8e63-3490a627f217)

As a content admin
1. Create a new Event at /node/add/event. Configure the event as shown in the 'Event Setup' screenshot above.
2. Save the Event
   - [ ] Validate that the start time for all events in MST begin at 10:00 am. This confirms the former Phoenix timezone issue is resolved.
3. Edit the Event
4. Click the 'Edit event series' button
5. Click the 'Edit' link in the 'Manage event series' modal for any of the events.
6. Change the start time for the event. Click 'Save'
    - [ ] Validate you are returned to the 'Manage event series' modal.
7. Close the modal.
8. Save the Event node.
   - [ ] Validate the node saved as expected.
 
### Definition of Done


### Select Team for PR review
